aws/resource_aws_ecs_task_definition.go

```diff
@@ -108,6 +108,44 @@ func resourceAwsEcsTaskDefinition() *schema.Resource {
 							Optional: true,
 							ForceNew: true,
 						},
+
+						"docker_volume_configuration": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"scope": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Computed: true,
+										ForceNew: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											ecs.ScopeShared,
+											ecs.ScopeTask,
+										}, false),
+									},
+									"autoprovision": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"driver": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"driver_opts": {
+										Type:     schema.TypeMap,
```

Comment:
While the schema validator errantly allows it in Terraform 0.11, we should always set `Elem` with `schema.TypeMap`, e.g.

```
Elem: &schema.Schema{Type: schema.TypeString},
```

See also: https://github.com/terraform-providers/terraform-provider-aws/pull/5350